src/validations/passport-gated/index.ts

```diff
@@ -36,17 +36,27 @@ const stampCredentials = STAMPS.map((stamp) => {
 // Useful to get stamp metadata and update `stampsMetata.json`
 // console.log('stampCredentials', JSON.stringify(stampCredentials.map((s) => ({"const": s.id, title: s.name}))));
 
-function hasValidIssuanceAndExpiration(credential: any, proposalTs: string) {
+function hasValidIssuanceAndExpiration(
+  credential: any,
+  proposalTs: string,
+  checkExpired: boolean
+) {
   const issuanceDate = Number(
     new Date(credential.issuanceDate).getTime() / 1000
   ).toFixed(0);
   const expirationDate = Number(
     new Date(credential.expirationDate).getTime() / 1000
   ).toFixed(0);
-  if (issuanceDate <= proposalTs && expirationDate >= proposalTs) {
-    return true;
+
+  let isValid = false;
+  if (issuanceDate <= proposalTs) {
+    if (checkExpired && expirationDate >= proposalTs) {
+      isValid = true;
+    } else if (~checkExpired) {
+      isValid = true;
```

Comment:
Right, but isValid is set to `false` above outside the conditional.

So `isValid` will become `true` only if either:
1. `checkExpired` is enabled and the stamp is not expired
2. `checkExpired` is false

And since this entire conditional is inside another conditional that checks whether the stamp was issued before the proposal started, `isValid` will remain false if the stamp was issued after the proposal started

Am I missing something?

Reply:
I mean consider this:

```typescript
let checkExpired = true
if (~checkExpired) console.log('CheckExpired is true');
checkExpired = false 
if (~checkExpired) console.log('CheckExpired is false');
```

It will print both 
```
CheckExpired is true
CheckExpired is false
```

Maybe you have added `~` as typo instead of `!`?

Reply:
Ah yes! I meant that to be `!` 🤦